src/openrct2/object/ObjectRepository.cpp

```diff
@@ -419,6 +419,23 @@ class ObjectRepository final : public IObjectRepository
         {
             conflict = FindObject(item.Identifier);
         }
+        // When there is a conflict between a DAT file and a JSON file, the JSON should take precedence.
+        if (conflict != nullptr && item.Generation == ObjectGeneration::JSON && conflict->Generation == ObjectGeneration::DAT)
+        {
+            for (size_t i = 0; i < _items.size(); i++)
```

Comment:
Can you use `conflict->Id`  to get the index rather than having to loop through the list?

Reply:
I’ll try that.